tools/OutlineService/OutlineService/OutlineService.cs

```diff
@@ -254,52 +261,35 @@ public partial class OutlineService : ServiceBase {
       }
-      eventLog.WriteEntry("Configured low TAP device metric", EventLogEntryType.Information);
 
       // Proxy routing: the proxy's IP address should be the only one that bypasses the router.
       // Save the best interface index for the proxy's address before we add the route. This
       // is necessary for updating the proxy route when the network changes; otherwise we get the
       // TAP device as the best interface.
       try {
-        GetSystemIpv4Gateway();
+        var systemGateway = GetSystemIpv4Gateway();
+        SetGatewayProperties(systemGateway);
       } catch (Exception e) {
-        throw new UnsupportedRoutingTableException($"unsupported routing table: {e.Message}");
-      }
-
-      try {
-        AddProxyRoute(proxyIp, gatewayIp.ToString(), gatewayInterfaceName);
-      } catch (Exception e) {
-        throw new Exception($"could not add route to proxy server: {e.Message}");
-      }
-
-      // Save the proxy's IP so we can reset routing.
-      this.proxyIp = proxyIp;
-
-      try {
-        // Route IPv4 traffic through the router. Instead of deleting the default IPv4 gateway (0.0.0.0/0),
-        // we resort to creating two more specific routes (see IPV4_SUBNETS) that take precedence over the
-        // default gateway. This way, we need not worry about the default gateway being recreated with a lower
-        // metric upon device sleep. This 'hack' was inspired by OpenVPN;
-        // see https://github.com/OpenVPN/openvpn3/commit/d08cc059e7132a3d3aee3dcd946fce4c35b1ced3#diff-1d76f0fd7ec04c6d1398288214a879c5R358.
-        foreach (string subnet in IPV4_SUBNETS) {
-          RunCommand(CMD_NETSH, $"interface ipv4 add route {subnet} nexthop={routerIp} interface={TAP_DEVICE_NAME} metric=0");
+        // Allow the connection to proceed if there is no network connectivity during auto connect.
+        if (!isAutoConnect || !(e is NoDefautlGatewayFoundException)) {
+          throw new UnsupportedRoutingTableException($"unsupported routing table: {e.Message}");
         }
-      } catch (Exception e) {
-        throw new Exception($"could not change default gateway: {e.Message}");
       }
 
-      try {
-        // Outline does not currently support IPv6, so we resort to disabling it while the VPN is active to
-        // prevent leakage. Removing the default IPv6 gateway is not enough since it gets re-created
-        // through router advertisements and DHCP (disabling these or IPv6 routing altogether requires a
-        // system reboot). Thus, we resort to creating three IPv6 routes (see IPV6_SUBNETS) to the loopback
-        // interface that are more specific than the default route, causing IPv6 traffic to get dropped.
-        foreach (string subnet in IPV6_SUBNETS) {
-          RunCommand(CMD_NETSH, $"interface ipv6 add route {subnet} interface={NetworkInterface.IPv6LoopbackInterfaceIndex} metric=0");
+      if (gatewayIp != null) {

```

Comment:
To clarify: if there is no network connectivity, we leave the IPv4 routing table as-is?

Reply:
Yes, I've improved the comment.